test/sqrt.jl

```diff
@@ -0,0 +1,44 @@
+using PDMats
+using Test
+using LinearAlgebra: LinearAlgebra
+
+_randPDMat(T, n) = (X = randn(T, n, n); PDMat(X * X' + LinearAlgebra.I))
+_randPDiagMat(T, n) = PDiagMat(rand(T, n))
+_randScalMat(T, n) = ScalMat(n, rand(T))
+_randPDSparseMat(T, n) = (X = T.(sprand(n, 1, 0.5)); PDSparseMat(X * X' + LinearAlgebra.I))
+
+function _pd_compare(A::AbstractPDMat, B::AbstractPDMat)
+    @test dim(A) == dim(B)
+    @test Matrix(A) ≈ Matrix(B)
+    @test cholesky(A).L ≈ cholesky(B).L
+    @test cholesky(A).U ≈ cholesky(B).U
+end
+
+function _pd_sqrt_compare(A::AbstractPDMat)
+    PDAsqrt = sqrt(A)
+    PDAaqrt_dense = PDMat(sqrt(Matrix(A)))
+    _pd_compare(PDAsqrt, PDAaqrt_dense)
+end
+
+function _pdsparse_sqrt_compare(A::AbstractPDMat)
+    # specific method required for testing cholesky of a sparse matrix
+    Asqrt = sqrt(A)
+    Asqrt_dense = PDMat(sqrt(Matrix(A)))
+    @test dim(Asqrt) == dim(Asqrt_dense)
+    @test Matrix(Asqrt) ≈ Matrix(Asqrt_dense)
+    cAsqrt = cholesky(Asqrt)
+    perm = cAsqrt.p  # account for permutation
+    L = sparse(cAsqrt.L)
+    @test Asqrt[perm, perm] ≈ L * L'
+end
```

Comment:
Is it possible to use https://github.com/JuliaStats/PDMats.jl/blob/84df00b41da9573c3e0b7e79bdcde2dda24bb2d5/test/testutils.jl#L95-L121 etc. here?

Reply:
the tests now rely entirely on these